homeassistant/components/wemo/__init__.py

```diff
@@ -33,13 +36,24 @@
 def coerce_host_port(value):
     """Validate that provided value is either just host or host:port.
 
+    If a hostname is provided, resolve it to an IP address before proceeding.
+
     Returns (host, None) or (host, port) respectively.
     """
     host, _, port = value.partition(":")
 
     if not host:
         raise vol.Invalid("host cannot be empty")
 
+    # Force hostnames into IP addresses for PyWemo compatibility
+    try:
+        ip_address(host)
```

Comment:
Config schemas are evaluated in the event loop. This means that you're now doing I/O in the event loop. That's not allowed.